apis/manifests/v1alpha1/packagemanifestlock_types.go

```diff
@@ -0,0 +1,71 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/utils/strings/slices"
+)
+
+// +kubebuilder:object:root=true
+type PackageManifestLock struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec PackageManifestLockSpec `json:"spec,omitempty"`
+}
+
+type PackageManifestLockSpec struct {
+	// List of resolved images
+	Images []PackageManifestLockImage `json:"images"`
+}
+
+// PackageManifestLockImage contains information about a resolved image
+type PackageManifestLockImage struct {
+	// Image name to be use to reference it in the templates
+	Name string `json:"name"`
+	// Image identifier (REPOSITORY[:TAG])
+	Image string `json:"image"`
+	// Image digest
+	Digest string `json:"digest"`
+}
+
+// Validates the PackageManifestLock and returns an aggregated list of all errors.
+func (m *PackageManifestLock) Validate() error {
+
+	var allErrs field.ErrorList
+
+	specImages := field.NewPath("spec").Child("images")
+	if len(m.Spec.Images) > 0 {
+		existingNames := []string{}
+		for i, image := range m.Spec.Images {
+			if len(image.Name) < 1 {
+				allErrs = append(allErrs,
+					field.Invalid(specImages.Index(i).Child("name"), image.Name, "must be non empty"))
+			} else if slices.Contains(existingNames, image.Name) {
+				allErrs = append(allErrs,
+					field.Invalid(specImages.Index(i).Child("name"), image.Name, "must be unique"))
+			} else {
+				existingNames = append(existingNames, image.Name)
+			}
+
+			if len(image.Image) < 1 {
+				allErrs = append(allErrs,
+					field.Invalid(specImages.Index(i).Child("image"), image.Image, "must be non empty"))
+			}
+
+			if len(image.Digest) < 1 {
+				allErrs = append(allErrs,
+					field.Invalid(specImages.Index(i).Child("digest"), image.Digest, "must be non empty"))
+			}
+		}
+	}
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return allErrs.ToAggregate()
+}
+
+func init() {
```

Comment:
Nit: I personally would put the init function right below the imports since it can cause a lot of headache if overlooked (speaking from experience :D)

Reply:
I placed it here as in `packagemanifest_types.go` ...if we decide to move we should do it everywhere imho